### ISSUE TITLE
opengl32.dll wrapper does not build

### ISSUE THREAD

**Issue by ric96** [AndreRH/hangover#76]
I am running into this issue:
```
[ric@localhost hangover]$ ./build/wine-host/loader/wine build/qemu/x86_64-windows-user/qemu-x86_64.exe.so ~/Downloads/npp.7.8.9.Installer.exe 
0050:fixme:x11settings:X11DRV_ChangeDisplaySettingsEx Changing non-primary adapter L"\\\\.\\DISPLAY2" settings is currently unsupported.
Did not manage to load any host DLLs.
Failed to load host DLLs
```

**Comment by stefand** [comment]
This suggests that hangover was not built correctly. Run make (without parallelism) in the top level directory and see what error you get.

**Comment by ric96** [comment]
> This suggests that hangover was not built correctly. Run make (without parallelism) in the top level directory and see what error you get.

aah, silly me.
now i need to figure out this:
`make[1]: *** No rule to make target '/home/ric/hangover/qemu/crypto/random-platform.c', needed by 'crypto/random-platform.o'.  Stop.`

**Comment by stefand** [comment]
huh, this is weird... Check if everything in qemu/ is checked out, including submodules.

**Comment by ric96** [comment]
> huh, this is weird... Check if everything in qemu/ is checked out, including submodules.

looks like it. and the file exists as well

**Comment by ric96** [comment]
NVM that was just a path error, my bad. Here's the actual compilation error.
```
make[1]: Entering directory '/home/ric/hangover/build/dlls32/riched32'
make[1]: 'riched32.dll' is up to date.
make[1]: Leaving directory '/home/ric/hangover/build/dlls32/riched32'
ln -sf ../../../dlls32/iphlpapi/iphlpapi.dll build/qemu/x86_64-windows-user/qemu_guest_dll32/
make -C build/dlls32/iphlpapi iphlpapi.dll
make[1]: Entering directory '/home/ric/hangover/build/dlls32/iphlpapi'
make[1]: 'iphlpapi.dll' is up to date.
make[1]: Leaving directory '/home/ric/hangover/build/dlls32/iphlpapi'
ln -sf ../../../dlls32/opengl32/opengl32.dll build/qemu/x86_64-windows-user/qemu_guest_dll32/
make -C build/dlls32/opengl32 opengl32.dll
make[1]: Entering directory '/home/ric/hangover/build/dlls32/opengl32'
i686-w64-mingw32-gcc ../../../dlls/opengl32/opengl_ext.c -c -o opengl_ext_g.o -I../../../dlls/opengl32/../include -I../../../dlls/opengl32/../../qemu/include -DQEMU_CURRENT_DLL=DLL_OPENGL32 -O2 -DCONST_VTABLE -DGUEST_BIT=32 -DHOST_BIT=64 -D_WIN32_WINNT=0x0603 -fcommon -DQEMU_DLL_GUEST -I../../wine-guest32/include -I../../../wine/include
../../../dlls/opengl32/opengl_ext.c:4582:90: error: unknown type name 'GLeglClientBufferEXT'
 4582 | void WINAPI glBufferStorageExternalEXT( GLenum target, GLintptr offset, GLsizeiptr size, GLeglClientBufferEXT clientBuffer, GLbitfield flags )
      |                                                                                          ^~~~~~~~~~~~~~~~~~~~
../../../dlls/opengl32/opengl_ext.c:15697:59: error: unknown type name 'GLeglImageOES'
15697 | void WINAPI glEGLImageTargetTexStorageEXT( GLenum target, GLeglImageOES image, const GLint* attrib_list )
      |                                                           ^~~~~~~~~~~~~
../../../dlls/opengl32/opengl_ext.c:15730:64: error: unknown type name 'GLeglImageOES'
15730 | void WINAPI glEGLImageTargetTextureStorageEXT( GLuint texture, GLeglImageOES image, const GLint* attrib_list )
      |                                                                ^~~~~~~~~~~~~
../../../dlls/opengl32/opengl_ext.c:33765:1: warning: data definition has no type or storage class
33765 | GLVULKANPROCNV WINAPI glGetVkProcAddrNV( const GLchar *name )
      | ^~~~~~~~~~~~~~
../../../dlls/opengl32/opengl_ext.c:33765:1: warning: type defaults to 'int' in declaration of 'GLVULKANPROCNV' [-Wimplicit-int]
../../../dlls/opengl32/opengl_ext.c:33765:1: warning: 'stdcall' attribute only applies to function types [-Wattributes]
../../../dlls/opengl32/opengl_ext.c:33765:23: error: expected ',' or ';' before 'glGetVkProcAddrNV'
33765 | GLVULKANPROCNV WINAPI glGetVkProcAddrNV( const GLchar *name )
      |                       ^~~~~~~~~~~~~~~~~
../../../dlls/opengl32/opengl_ext.c:48203:95: error: unknown type name 'GLeglClientBufferEXT'
48203 | void WINAPI glNamedBufferStorageExternalEXT( GLuint buffer, GLintptr offset, GLsizeiptr size, GLeglClientBufferEXT clientBuffer, GLbitfield flags )
      |                                                                                               ^~~~~~~~~~~~~~~~~~~~
../../../dlls/opengl32/opengl_ext.c: In function 'glPathGlyphIndexRangeNV':
../../../dlls/opengl32/opengl_ext.c:51164:23: warning: assignment to 'uint64_t' {aka 'long long unsigned int'} from 'GLuint *' {aka 'unsigned int *'} makes integer from pointer without a cast [-Wint-conversion]
51164 |     call.baseAndCount = baseAndCount;
      |                       ^
../../../dlls/opengl32/opengl_ext.c: At top level:
../../../dlls/opengl32/opengl_ext.c:91051:61: error: 'glBufferStorageExternalEXT' undeclared here (not in a function); did you mean 'qemu_glBufferStorageExternalEXT'?
91051 |   { "glBufferStorageExternalEXT", "GL_EXT_external_buffer", glBufferStorageExternalEXT },
      |                                                             ^~~~~~~~~~~~~~~~~~~~~~~~~~
      |                                                             qemu_glBufferStorageExternalEXT
../../../dlls/opengl32/opengl_ext.c:91367:66: error: 'glEGLImageTargetTexStorageEXT' undeclared here (not in a function); did you mean 'qemu_glEGLImageTargetTexStorageEXT'?
91367 |   { "glEGLImageTargetTexStorageEXT", "GL_EXT_EGL_image_storage", glEGLImageTargetTexStorageEXT },
      |                                                                  ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
      |                                                                  qemu_glEGLImageTargetTexStorageEXT
../../../dlls/opengl32/opengl_ext.c:91368:70: error: 'glEGLImageTargetTextureStorageEXT' undeclared here (not in a function); did you mean 'qemu_glEGLImageTargetTextureStorageEXT'?
91368 |   { "glEGLImageTargetTextureStorageEXT", "GL_EXT_EGL_image_storage", glEGLImageTargetTextureStorageEXT },
      |                                                                      ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
      |                                                                      qemu_glEGLImageTargetTextureStorageEXT
../../../dlls/opengl32/opengl_ext.c:91908:53: error: 'glGetVkProcAddrNV' undeclared here (not in a function); did you mean 'qemu_glGetVkProcAddrNV'?
91908 |   { "glGetVkProcAddrNV", "GL_NV_draw_vulkan_image", glGetVkProcAddrNV },
      |                                                     ^~~~~~~~~~~~~~~~~
      |                                                     qemu_glGetVkProcAddrNV
../../../dlls/opengl32/opengl_ext.c:92338:66: error: 'glNamedBufferStorageExternalEXT' undeclared here (not in a function); did you mean 'qemu_glNamedBufferStorageExternalEXT'?
92338 |   { "glNamedBufferStorageExternalEXT", "GL_EXT_external_buffer", glNamedBufferStorageExternalEXT },
      |                                                                  ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
      |                                                                  qemu_glNamedBufferStorageExternalEXT
make[1]: *** [../../../dlls/opengl32/Makefile:18: opengl_ext_g.o] Error 1
make[1]: Leaving directory '/home/ric/hangover/build/dlls32/opengl32'
make: *** [Makefile:203: build/dlls32/opengl32/opengl32.dll] Error 2
[ric@localhost hangover]$ 
```

**Comment by stefand** [comment]
Huh, I might have done something wrong there, or there are problems with GL headers. I doubt it is the latter case because Wine has its own GL headers and we should be using that.

I won't have time to look into it in the next days. My quick suggestion is to use the touch command to create the output files until make in build/dlls32/opengl32 and build/dlls64/opengl32 succeeds, and then proceed with running make on the top level directory. You won't have a working OpenGL wrapper, but it is a relatively new thing and unlikely to work well anyway.

**Comment by ric96** [comment]
> Huh, I might have done something wrong there, or there are problems with GL headers. I doubt it is the latter case because Wine has its own GL headers and we should be using that.
> 
> I won't have time to look into it in the next days. My quick suggestion is to use the touch command to create the output files until make in build/dlls32/opengl32 and build/dlls64/opengl32 succeeds, and then proceed with running make on the top level directory. You won't have a working OpenGL wrapper, but it is a relatively new thing and unlikely to work well anyway.

ok that workaround works. Do you want me to close this and open a new issue for opengl32?

**Comment by stefand** [comment]
I've changed the title. I hope to have some time to look into it next week :-)

**Comment by AndreRH** [comment]
Was running into this on powerpc64 :)
Was fixed by 2e69e0eea77882cc054b74ed21a959950d1c52c0
